storage/atsd/atsd.go

```diff
@@ -0,0 +1,400 @@
+		if err != nil {
+			panic(err)
+		}
+		threshold = atsdStorageDriver.Percent(val)
+	} else {
+		val, err := strconv.ParseFloat(groupValues[2], 64)
+		if err != nil {
+			panic(err)
+		}
+		threshold = atsdStorageDriver.Absolute(val)
+	}
+	self[groupName] = atsdStorageDriver.DeduplicationParams{Interval: interval, Threshold: threshold}
+	return nil
+}
+
+// need to parse duration in toml configuration file
+type duration struct {
+	time.Duration
+}
+
+func (d *duration) UnmarshalText(text []byte) error {
+	var err error
+	d.Duration, err = time.ParseDuration(string(text))
+	return err
+}
+
+func newStorage(username, password, hostname string, argDbBufferDuration time.Duration) storage.StorageDriver {
+
+	innerStorageConfig := atsdStorageDriver.GetDefaultConfig()
+
+	selfConfig := struct {
+		Cadvisor struct {
+			IncludeAllMajorNumbers bool     `toml:"store_major_numbers"`
+			UserCgroupsEnabled     bool     `toml:"store_user_cgroups"`
+			PropertyInterval       duration `toml:"property_interval"`
+			SamplingInterval       duration `toml:"sampling_interval"`
+			DockerHost             string   `toml:"docker_host"`
+		}
+	}{}
+
+	if *configFilePath != "" {
+		_, err := toml.DecodeFile(*configFilePath, &innerStorageConfig)
+		if err != nil {
+			panic(err)
+		}
+		_, err = toml.DecodeFile(*configFilePath, &selfConfig)
+		if err != nil {
+			panic(err)
+		}
+	} else {
+		url, err := url.Parse(*urlString)
+		if err != nil {
+			panic(err)
+		}
+		innerStorageConfig.Protocol = *protocol
+		innerStorageConfig.DataReceiverHostport = *writeHost
+		innerStorageConfig.Url = url
+		innerStorageConfig.MemstoreLimit = *memstoreLimit
+		innerStorageConfig.ConnectionLimit = *connectionLimit
+		innerStorageConfig.GroupParams = deduplication
+		selfConfig.Cadvisor.DockerHost = *dockerHost
+		selfConfig.Cadvisor.PropertyInterval = duration{*propertyInterval}
+		selfConfig.Cadvisor.SamplingInterval = duration{*samplingInterval}
+		selfConfig.Cadvisor.IncludeAllMajorNumbers = *includeAllMajorNumbers
+		selfConfig.Cadvisor.UserCgroupsEnabled = *userCgroupsEnabled
+	}
+
+	selfMetricsEntity := selfConfig.Cadvisor.DockerHost + "/" + hostname
+	innerStorageConfig.Username = username
+	innerStorageConfig.Password = password
+	innerStorageConfig.SelfMetricEntity = selfMetricsEntity
+	innerStorageConfig.MetricPrefix = metricPrefix
+	innerStorageConfig.UpdateInterval = argDbBufferDuration
+
+	storageFactory := atsdStorageDriver.NewFactoryFromConfig(innerStorageConfig)
+	storage := &Storage{
+		machineName:             selfConfig.Cadvisor.DockerHost,
+		propertyInterval:        selfConfig.Cadvisor.PropertyInterval.Duration,
+		samplingInterval:        selfConfig.Cadvisor.SamplingInterval.Duration,
+		innerStorage:            storageFactory.Create(),
+		lastTimeSentPropertyMap: make(map[string]uint64),
+		lastTimeSentSeriesMap:   make(map[string]uint64),
+		firstTime:               make(map[string]bool),
```

Comment:
What do you think about collapsing these three maps into a single `map[string]sentStatus{....}`, where `sentStatus` (can you think of a better name) is a struct containing these 3 fields per-container?